### ISSUE TITLE
Move ModalRoute.of(context) to access page args out of build()

### ISSUE THREAD

**Issue by mrahimygk** [mrahimygk/kip-flutter#5]
There are various situations that we need passed page arguments to call apis, but they need to move out of build(). `initState()` would be a good location for them.

**Comment by mrahimygk** [comment]
@march93

Seems I need to explain the problem better.
We are parsing args in `build`. this could result in calling our imperative line of code e.g. `addCheckBox();` multiple times. To avoid this problem we used a global flag to check if the parsing is done already. But there can be a better way that considers the lifecycle of the view thus we can use that lifecycle function to init our NoteModel and do things like `addCheckBox();` in `thatLovelyMethod(context)` and just draw the UI in `build()`. The `didChangeDependencies()` method has our desired `context` available but I know that it can be called multiple times in the lifecycle and I cannot fill my `NoteModel`/`checkboxList` there.

**Comment by march93** [comment]
@mrahimygk makes sense! Do you have any suggestions on the best lifecycle method to use? When `args` is in `initState`,  `build` isn't able to find it as `initState` may not have completed yet. `didChangeDependencies` runs after `initState` completes but can't be used in this case

**Comment by mrahimygk** [comment]
I saw these 3 samples in stackoverflow:

```
Future.delayed(Duration.zero,() {
  parseArgs(context)
}
```

-=-=-=-=-=-

```
SchedulerBinding.instance.addPostFrameCallback((_) {
  parseArgs(context)
});
```

-=-=-=-=-
```
() async {
  await Future.delayed(Duration.zero);
  parseArgs(context)
}();
```

**Comment by hiashutoshsingh** [comment]
I have been using top 2 solutions for a while, it works like a charm. 
Anything you want to do with context can be done with above approaches.

**Comment by march93** [comment]
@mrahimygk I just remembered another thing I used to do when initializing data. What do you think of `args ??= ModalRoute.of(context).settings.arguments as AddNotePageArguments;` initialized in `didChangeDependencies`? I tried the above examples but they don't work in `initState` as the context is not available when it's first called.

**Comment by mrahimygk** [comment]
We have to keep variables like `hasAddedNewCheckboxAlready` and they just will be moved to `didChangeDependencies`. So I presume this won't fix our main issue.

**Comment by hiashutoshsingh** [comment]
> @mrahimygk I just remembered another thing I used to do when initializing data. What do you think of `args ??= ModalRoute.of(context).settings.arguments as AddNotePageArguments;` initialized in `didChangeDependencies`? I tried the above examples but they don't work in `initState` as the context is not available when it's first called.

did you user above methods like wrapping your function in delay ? I think it should work it works in my case.

**Comment by march93** [comment]
@hiashutoshsingh yea I wrapped it in delay, would you be able to share that part of the code?

**Comment by Trung15010802** [comment]
any update?
